packages/features/bookings/Booker/components/hooks/useInitialFormValues.ts

```diff
@@ -136,11 +139,14 @@ export function useInitialFormValues({
     name,
     username,
     prefillFormParams,
+    prevResponse,
   ]);
 
   // When initialValues is available(after doing async schema parsing) or session is available(so that we can prefill logged-in user email and name), we need to reset the form with the initialValues
   // We also need the key to change if the bookingId changes, so that the form is reset and rerendered with the new initialValues
-  const key = `${Object.keys(initialValues).length}_${hasSession ? 1 : 0}_${initialValues?.bookingId ?? 0}`;
+  const key = `${prevResponse ? JSON.stringify(initialValues).length : Object.keys(initialValues).length}_${
```

Comment:
When previous responses were fetched, key was immediately updated and useEffect was executed before the initalValues could update. Not sure if this is the best way